libraries/rush-sdk/src/index.ts

```diff
@@ -84,3 +126,43 @@ for (const property in rushLibModule) {
     });
```

Comment:
When the above code fails, it does `throw new Error()`.

I envisioned that the CLI tool might look like this:
```ts
import { RushConfiguration } from  '@rushstack/rush-sdk';

console.log('Doing something useful...');
```

In which case the `import` fails with an uncaught exception:
```ts
C:\MyRepo>helpful-tool
rush-sdk\lib\index.js:104
    throw new Error('The "@rushstack/rush-sdk" package context has not been initialized.');
    ^

Error: The "@rushstack/rush-sdk" package context has not been initialized.
    at Object.<anonymous> (C:\Git\rushstack\libraries\rush-sdk\lib\index.js:104:11)        
    at Module._compile (internal/modules/cjs/loader.js:1068:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1097:10)
```

This isn't great however.  There's a couple ways we could improve it:

1. `rush-sdk` does `console.error()` and then `process.exit(1)`.  Generally `process.exit()` should be avoided but it may be okay here.
2. The caller must call `try { require('rush-sdk') }` in the entry point for their tool.  They can also use regular `import { ... } from 'rush-sdk'` in other files, but they MUST ensure that those imports are evaluated after the entry point has loaded the SDK.

Which is better?